src/main/java/com/ait/lienzo/client/core/shape/TextTruncateWrapper.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.shape;
+
+import java.util.ArrayList;
+
+import com.ait.lienzo.client.core.Context2D;
+import com.ait.lienzo.client.core.types.BoundingBox;
+
+/**
+ * ITextWrapper implementation that truncates text and appends "..." if there is no space left.
+ */
+@SuppressWarnings("Duplicates")
+public class TextTruncateWrapper extends TextNoWrap {
+
+    private BoundingBox wrapBoundaries;
+    private final double Y_OFFSET = 0.8;
+
+    public TextTruncateWrapper(final Text text,
+                               final BoundingBox wrapBoundaries) {
+        super(text);
+        setWrapBoundaries(wrapBoundaries);
+    }
+
+    public BoundingBox getWrapBoundaries() {
+        return wrapBoundaries;
+    }
+
+    public TextTruncateWrapper setWrapBoundaries(final BoundingBox boundaries) {
+        wrapBoundaries = boundaries;
+        return this;
+    }
+
+    @Override
+    public BoundingBox getBoundingBox() {
+        final double[] boundaries = calculateWrapBoundaries();
+        return new BoundingBox().addX(0).addX(wrapBoundaries.getWidth()).addY(0).addY(boundaries[1]);
+    }
+
+    private double getWrapBoundariesWidth() {
+        final double margin = 4 * getBoundingBoxForString(" ").getWidth();
```

Comment:
`getBoundBoxForString(" ")` is quite an expensive operation; perhaps this could be moved to a static (or at least single re-used instance)? The method `getWrapBoundardiesWidth()` is called by `getBoundingBox()` that _may_ be called quite a lot (I'm guessing).. but either way re-calculating the width of a " "  to add a margin to the text seems unnecessary.

Reply:
Oh, you'll also have seen the `getLineHeight()`(?) method that uses "Mg". This is because "M" is the tallest letter and "g" has the lowest _descender_ (hook/curve of the bottom of the "g"). Perhaps using the width of a single "M" would be better than 4 spaces?

Reply:
Agree, I'm changing it now.